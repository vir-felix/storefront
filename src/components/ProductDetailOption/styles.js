export default (theme) => ({
  variantOptionContainer: {
    display: "flex",
    width: `${theme.spacing.unit * 3.125}%`,
    maxWidth: `${theme.spacing.unit * 3.125}%`,
    padding: theme.spacing.unit * 1.25
  },
  optionButton: {
    "flex": "1 1 auto",
    "fontWeight": 600,
    "padding": theme.spacing.unit * 1.25,
    "borderRadius": "2px",
    "backgroundColor": theme.palette.primary.contrastText,
    "border": "1px solid",
    "borderColor": theme.palette.primary.borderColor,
    "&:hover": {
      backgroundColor: theme.palette.reaction.activeElementBackground
    }
  },
  optionText: {
    fontWeight: 500,
    fontSize: "14px"
  },
  isSelected: {
    backgroundColor: theme.palette.reaction.activeElementBackground
  }
});
