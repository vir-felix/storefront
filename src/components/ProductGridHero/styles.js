export default (theme) => ({
  heroImg: {
    width: "100%",
    height: "325px",
    objectFit: "cover"
  },
  productGridContainer: {
    maxWidth: theme.grid.productGridMaxWidth,
    marginLeft: "auto",
    marginRight: "auto"
  }
});
