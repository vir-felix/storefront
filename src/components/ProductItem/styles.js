export const styles = (theme) => ({
  "@keyframes spin": {
    from: { transform: "rotate(0deg)" },
    to: { transform: "rotate(-360deg)" }
  },
  "root": {},
  "productInfo": {
    display: "flex",
    justifyContent: "space-between",
    alignItems: "center",
    paddingTop: 10
  },
  "productMedia": {
    backgroundColor: theme.palette.primary.contrastText,
    position: "relative"
  },
  "img": {
    height: "auto",
    width: "100%"
  },
  "imgLoading": {
    alignItems: "center",
    backgroundColor: "transparent",
    display: "flex",
    fontSize: 48,
    height: 100,
    justifyContent: "center",
    left: "calc(50% - 50px)",
    position: "absolute",
    top: "calc(50% - 50px)",
    width: 100
  },
  "loadingIcon": {
    color: theme.palette.primary.light,
    fontSize: "inherit",
    animationName: "spin",
    animationDuration: theme.transitions.duration.standard * 2,
    animationTimingFunction: theme.transitions.easing.sharp,
    animationIterationCount: "infinite",
    animationFillMode: "both"
  },
  "link": {
    ...theme.typography.body2,
    "textAlign": "left",
    "&:hover": {
      color: theme.palette.secondary.main
    }
  }
});
