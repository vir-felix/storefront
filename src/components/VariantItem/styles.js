export default (theme) => ({
  variantButton: {
    "display": "flex",
    "justifyContent": "space-between",
    "textTransform": "none",
    "width": "100%",
    "borderRadius": "2px",
    "backgroundColor": theme.palette.primary.contrastText,
    "border": "1px solid",
    "borderColor": theme.palette.primary.contrastText,
    "paddingTop": theme.spacing.unit * 1.25,
    "paddingBottom": theme.spacing.unit * 1.25,
    "paddingLeft": theme.spacing.unit * 2.5,
    "paddingRight": theme.spacing.unit * 2.5,
    "&:hover": {
      border: "1px solid",
      borderColor: theme.palette.reaction.activeElementBorderColor,
      backgroundColor: theme.palette.primary.contrastText
    }
  },
  activeVariant: {
    border: "1px solid",
    borderColor: theme.palette.reaction.activeElementBorderColor
  }
});
