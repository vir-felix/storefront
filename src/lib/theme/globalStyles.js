export default
<style global jsx>
  {`
    a {
      color: inherit;
      text-decoration: none;
    }
  `}
</style>;
