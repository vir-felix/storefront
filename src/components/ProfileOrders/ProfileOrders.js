import React, { Component } from "react";
import PropTypes from "prop-types";
import { inject, observer } from "mobx-react";
import { Typography } from "@material-ui/core";
import { withStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import relayConnectionToArray from "lib/utils/relayConnectionToArray";
import OrderCard from "components/OrderCard";
import Select from "components/Select";
import withOrders from "containers/order/withOrders";
import ErrorPage from "../../pages/_error";

const styles = (theme) => ({
  profileOrdersContainer: {},
  profileOrdersList: {},
  profileOrdersSelect: {
    marginBottom: theme.spacing.unit * 4
  },
  profileOrdersTitle: {
    marginBottom: theme.spacing.unit * 4
  }
});

@withStyles(styles, { withTheme: true })
@inject("uiStore")
@withOrders
@observer
class ProfileOrders extends Component {
  static propTypes = {
    classes: PropTypes.object,
    orders: PropTypes.arrayOf(PropTypes.shape({
      email: PropTypes.string.isRequired,
      fulfillmentGroups: PropTypes.arrayOf(PropTypes.object).isRequired,
      payments: PropTypes.arrayOf(PropTypes.object),
      referenceId: PropTypes.string.isRequired
    })),
    shop: PropTypes.shape({
      name: PropTypes.string.isRequired,
      description: PropTypes.string
    }),
    uiStore: PropTypes.shape({
      accountProfileOptions: PropTypes.shape({
        orderStatusQuery: PropTypes.string
      }),
      setAccountProfileOrderStatusQueryVariable: PropTypes.func
    }).isRequired
  };

  handleOrderStatusTypeChange = (event) => {
    const { uiStore } = this.props;
    uiStore.setAccountProfileOrderStatusQueryVariable(event.target.value);
  };

  renderOrderTypeSelect() {
    const { uiStore: { accountProfileOptions: { orderStatusQuery } } } = this.props;

    const orderStatusOptions = [
      {
        name: "All",
        value: "all"
      },
      {
        name: "Open orders",
        value: "open"
      },
      {
        name: "Completed orders",
        value: "completed"
      },
      {
        name: "Canceled orders",
        value: "canceled"
      }
    ];

    return (
      <Select
        value={orderStatusQuery}
        options={orderStatusOptions}
        inputProps={{
          name: "orderStatus",
          id: "order-status"
        }}
        onChange={this.handleOrderStatusTypeChange}
      />
    );
  }

  renderOrders() {
    const { orders } = this.props;

    // Use relayConnectionToArray to remove edges / nodes levels from orders object
    const ordersToArray = (orders && relayConnectionToArray(orders)) || [];

    return ordersToArray.map((order) => <OrderCard order={order} />);
  }

  render() {
    const { classes, orders, shop } = this.props;

    if (!orders) return <ErrorPage shop={shop} subtitle="No orders found" />;

    return (
      <Grid className={classes.profileOrdersContainer} container>
        <Grid className={classes.profileOrdersTitle} item xs={12} md={12}>
          <Typography variant="h6">Orders</Typography>
        </Grid>
        <Grid className={classes.profileOrdersSelect} item xs={12} md={12}>
          {this.renderOrderTypeSelect()}
        </Grid>
        <Grid className={classes.profileOrdersList} item xs={12} md={12}>
          {this.renderOrders()}
        </Grid>
      </Grid>
    );
  }
}

export default ProfileOrders;
