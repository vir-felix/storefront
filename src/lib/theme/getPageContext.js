import { SheetsRegistry } from "jss";
import { createGenerateClassName } from "@material-ui/core/styles";
import theme from "./reactionTheme";

/**
 * @summary Creates and returns a page context object
 * @returns {Object} Page context object
 */
function createPageContext() {
  return {
    theme,
    // This is needed in order to deduplicate the injection of CSS in the page.
    sheetsManager: new Map(),
    // This is needed in order to inject the critical CSS.
    sheetsRegistry: new SheetsRegistry(),
    // The standard class name generator.
    generateClassName: createGenerateClassName()
  };
}

/**
 * Either creates a new or returns an existing page context object. If called from Node, creates
 * a new object every time. Otherwise creates and caches an object in the browser and
 * subsequent calls will return that.
 *
 * @summary Gets a page context object
 * @returns {Object} Page context object
 */
export default function getPageContext() {
  // Make sure to create a new context for every server-side request so that data
  // isn't shared between connections (which would be bad).
  if (!process.browser) {
    return createPageContext();
  }

  // Reuse context on the client-side.
  if (!global.__INIT_MATERIAL_UI__) {
    global.__INIT_MATERIAL_UI__ = createPageContext();
  }

  return global.__INIT_MATERIAL_UI__;
}
