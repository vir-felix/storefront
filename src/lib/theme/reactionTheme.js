import { createMuiTheme } from "material-ui/styles";

const theme = createMuiTheme({
  palette: {
    primary: {
      light: "#26B0F9",
      main: "#1999DD",
      dark: "#172F3C",
      contrastText: "#FFFFFF",
      borderColor: "#CCCCCC"
    },
    secondary: {
      light: "#5d8ea9",
      main: "#5E7480",
      dark: "#1D1D1D",
      contrastText: "#000000"
    },
    error: {
      light: "#E54F5D",
      main: "#CD3F4C",
      dark: "#3C1F21",
      contrastText: "#FFFFFF"
    },
    reaction: {
      activeElementBorderColor: "#94E8D1",
      activeElementBackground: "#E6E6E6"
    }

  },
  typography: {
    fontFamily: "Source Sans Pro, Helvetica Neue, Helvetica, sans-serif",
    fontSize: 16,
    fontWeightLight: 200,
    fontWeightRegular: 400,
    fontWeightMedium: 700,
    fontWeightBold: 700
  }
});

export default theme;
