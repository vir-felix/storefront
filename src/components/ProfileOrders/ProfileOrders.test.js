import React from "react";
import renderer from "react-test-renderer";
import { ComponentsProvider } from "@reactioncommerce/components-context";
import { MuiThemeProvider } from "@material-ui/core/styles";
import theme from "custom/reactionTheme";
import mockComponents from "../../../tests/mockComponents";
import ProfileOrders from "./ProfileOrders";

const orders = [];

test("basic snapshot", () => {
  const component = renderer.create((
    <ComponentsProvider value={mockComponents}>
      <MuiThemeProvider theme={theme}>
        <ProfileOrders
          orders={orders}
        />
      </MuiThemeProvider>
    </ComponentsProvider>
  ));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
