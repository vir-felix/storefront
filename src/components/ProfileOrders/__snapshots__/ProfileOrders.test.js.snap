// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
<div
  className="Error-root-5"
>
  <p
    className="MuiTypography-root-8 MuiTypography-body1-17 MuiTypography-paragraph-36 Error-errorMessage-6"
  >
    Sorry! We couldn't find what you're looking for.
  </p>
  <p
    className="MuiTypography-root-8 MuiTypography-body1-17 Error-errorLink-7"
  >
    <a
      className="SkLink-anchor-44"
      href="/"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="link"
      tabIndex={0}
    >
      Home
    </a>
  </p>
</div>
`;
